### PR TITLE
[howstuffwoks] fix _VALID_URL regex

### PR DESCRIPTION
**Pull request by remitamine**
make the extractor works with url that does not contain a number in the url like that:
http://shows.howstuffworks.com/stuff-to-blow-your-mind/optical-illusions-video.htm

### CODE REVIEW

**Comment by dstftw** [comment]
Can you add a test?

File: youtube_dl/extractor/howstuffworks.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 
 
 class HowStuffWorksIE(InfoExtractor):
-    _VALID_URL = r'https?://[\da-z-]+\.howstuffworks\.com/(?:[^/]+/)*\d+-(?P<id>.+?)-video\.htm'
+    _VALID_URL = r'https?://[\da-z-]+\.howstuffworks\.com/(?:[^/]+/)*(\d+-)*(?P<id>.+?)-video\.htm'
```
Comment (dstftw) [inline youtube_dl/extractor/howstuffworks.py]: That's wrong by the way. It matches zero on more `\d+-` parts. Also do not capture group if you dont not use it.

**Comment by remitamine** [comment]
i correct the mistakes you point at is this enough.

File: youtube_dl/extractor/howstuffworks.py [hunk 2]
```diff
@@ -46,6 +46,17 @@ class HowStuffWorksIE(InfoExtractor):
                 'thumbnail': 're:^https?://.*\.jpg$',
             },
         },
+        {
+            'url': 'http://shows.howstuffworks.com/stuff-to-blow-your-mind/optical-illusions-video.htm',
+            'info_dict': {
```
Comment (dstftw) [inline youtube_dl/extractor/howstuffworks.py]: `'only_matching': True` is enough.

**Pull request merged**